src/ffpuppet/profile.py

```diff
@@ -0,0 +1,265 @@
+        assert self.path
+        if isinstance(extension, (list, tuple)):
+            extensions = [Path(x) for x in extension]
+        else:
+            extensions = [Path(extension)]
+        ext_path = self.path / "extensions"
+        ext_path.mkdir(exist_ok=True)
+        for ext in extensions:
+            if ext.is_file() and ext.name.endswith(".xpi"):
+                copyfile(ext, ext_path / ext.name)
+            elif ext.is_dir():
+                # read manifest to see what the folder should be named
+                ext_name = None
+                if (ext / "manifest.json").is_file():
+                    try:
+                        with (ext / "manifest.json").open("r") as manifest:
+                            manifest_loaded_json = json_load(manifest)
+                        ext_name = manifest_loaded_json["applications"]["gecko"]["id"]
+                    except (OSError, KeyError, ValueError) as exc:
+                        LOG.debug("Failed to parse manifest.json: %s", exc)
+                elif (ext / "install.rdf").is_file():
+                    try:
+                        xmlns = {
+                            "x": "http://www.w3.org/1999/02/22-rdf-syntax-ns#",
+                            "em": "http://www.mozilla.org/2004/em-rdf#",
+                        }
+                        tree = ElementTree.parse(str(ext / "install.rdf"))
+                        assert tree.getroot().tag == f"{{{xmlns['x']}}}RDF"
+                        ids = tree.findall("./x:Description/em:id", namespaces=xmlns)
+                        assert len(ids) == 1
+                        ext_name = ids[0].text
+                    except (AssertionError, OSError, ElementTree.ParseError) as exc:
+                        LOG.debug("Failed to parse install.rdf: %s", exc)
+                if ext_name is None:
+                    raise RuntimeError(
+                        f"Failed to find extension id in manifest: '{ext}'"
+                    )
+                copytree(ext, self.path / "extensions" / ext_name)
+            else:
+                raise RuntimeError(f"Unknown extension: '{ext}'")
+
+    def _copy_prefs_file(self, prefs_file: str) -> None:
+        assert self.path
+        LOG.debug("using prefs.js: %r", prefs_file)
+        copyfile(prefs_file, self.path / "prefs.js")
+        self._add_times_json()
+
+    def _copy_template(self, template: str) -> None:
+        assert self.path
+        LOG.debug("using profile template: %r", template)
+        rmtree(self.path)
+        copytree(template, self.path)
+        invalid_prefs = self.path / "Invalidprefs.js"
+        # if Invalidprefs.js was copied from the template profile remove it
+        if invalid_prefs.is_file():
+            invalid_prefs.unlink()
+
+    def _install_cert(self, cert_file: str) -> None:
+        assert self.path
+        LOG.debug("installing certificate: %r", cert_file)
+        try:
+            check_output(
```

Comment:
I'm guessing the use-case here is adding a single cert only. But this call hangs the second time it's run (asking for password). Maybe skip it if it looks like we already have an nss db in `self.path`?